ios/Gekidou/Sources/Gekidou/Keychain.swift

```diff
@@ -71,35 +76,63 @@ public class Keychain: NSObject {
         guard certificate != nil else {
             throw KeychainError.CertificateForIdentityNotFound
         }
-        
+
         return (identity, certificate!)
     }
-    
-    @objc public func getTokenObjc(for serverUrl: String) -> String? {
-        return try? getToken(for: serverUrl)
+
+    @objc public func getCredentialsObjc(for serverUrl: String) -> NSDictionary? {
+        guard let credentials = try? getCredentials(for: serverUrl) else { return nil }
+        return [
+            "token": credentials.token as Any,
+            "preauthSecret": credentials.preauthSecret as Any
+        ]
+    }
+
+    public func getCredentials(for serverUrl: String) throws -> ServerCredentials? {
+        // Get main token from serverUrl key
+        let token = try getMainToken(for: serverUrl)
+        let preauthSecret = try? getPreauthSecret(for: serverUrl)
+
+        return ServerCredentials(token: token, preauthSecret: preauthSecret)
     }
-    
-    public func getToken(for serverUrl: String) throws -> String? {
+
+    private func getMainToken(for serverUrl: String) throws -> String? {
         var attributes = try buildTokenAttributes(for: serverUrl)
         attributes[kSecMatchLimit] = kSecMatchLimitOne
         attributes[kSecReturnData] = kCFBooleanTrue
-        
+
         var result: AnyObject?
         let status = SecItemCopyMatching(attributes as CFDictionary, &result)
         let data = result as? Data
         if status == errSecSuccess && data != nil {
-            let token = String(data: data!, encoding: .utf8)
-            return token
+            let credentialsString = String(data: data!, encoding: .utf8)
+            return credentialsString
         }
-        
+
```

Comment:
0/5 to avoid unneeded diff, we could revert this variable name change, but not sure if the variable name change was intentional.

Reply:
Reverted: https://github.com/mattermost/mattermost-mobile/pull/9082/commits/024ebfcd874ba7a22d968f69bbe28982c23a8faa